File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/Command.java
package com.example.try_gameengine.remotecontroller;

import android.view.MotionEvent;

import com.example.try_gameengine.remotecontroller.RemoteController.CommandType;

/**
 * {@code Command} is a default command handler for 4D(4 Direction keys
, total 4 keys.
 * 
 * @author irons
 */
public interface Command {
	/**
	 * @param x
	 * @param y
	 * @param event
	 * @return
	 */
	public boolean checkExecute(float x, float y, MotionEvent event);

	/**
	 * @return
	 */
	public CommandType execute();

	/**
	 * @param pointerId
	 */
	public void setMotionEventPointerId(int pointerId);

	/**
	 * @return
	 */
	public int getMotionEventPointerId();
}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/Command.kt
package com.example.try_gameengine.remotecontroller

import android.view.MotionEvent

/**
 * `Command` is a default command handler for 4D(4 Direction keys
 * , total 4 keys.
 * 
 * @author irons
 // */
interface Command {
    /**
     * @param x
     * @param y
     * @param event
     * @return
     // */
    fun checkExecute(x: Float, y: Float, event: MotionEvent?): Boolean

    /**
     * @return
     // */
    fun execute(): RemoteController.CommandType?

    /**
     * @param pointerId
     // */
    fun setMotionEventPointerId(pointerId: Int)

    /**
     * @return
     // */
    fun getMotionEventPointerId(): Int
}
